### PR TITLE
[ticket/14739] Remove SQLite 2.8.x database driver

### PR DESCRIPTION
**Pull request by senky**
Checklist:

- [x] Correct branch: master for new features; 3.2.x, 3.1.x for fixes
- [x] Tests pass
- [x] Code follows coding guidelines: [master / 3.2.x](https://area51.phpbb.com/docs/master/coding-guidelines.html), [3.1.x](https://area51.phpbb.com/docs/31x/coding-guidelines.html)
- [x] Commit follows commit message [format](https://wiki.phpbb.com/Git#Commit_Messages)

Tracker ticket (set the ticket ID to **your ticket ID**):

https://tracker.phpbb.com/browse/PHPBB3-14739

PHPBB3-14739

### CODE REVIEW

File: phpBB/phpbb/search/fulltext_native.php [hunk 1]
```diff
@@ -1202,7 +1201,7 @@ public function author_search($type, $firstpost_only, $sort_by_sql, $sort_key, $
 								$post_visibility
 								$sql_fora
 								AND t.topic_id = p.topic_id
-								$sql_time" . (($this->db->get_sql_layer() == 'sqlite' || $this->db->get_sql_layer() == 'sqlite3') ? ')' : '');
+								$sql_time" . (($this->db->get_sql_layer() == 'sqlite3') ? ')' : '');
```
Comment (Nicofuma) [inline phpBB/phpbb/search/fulltext_native.php]: all parenthesis are useless: `$sql_time" . $this->db->get_sql_layer() == 'sqlite3' ? ')' : '';`

Comment (senky) [inline phpBB/phpbb/search/fulltext_native.php]: Not true, you need to enclose expression in brackets to return it's value. Try it yourself here:
http://sandbox.onlinephpfunctions.com/
try to execute this:
```php
<?php
$string = "text" . 'a' == 'a' ? '-append' : '';
echo $string;
```
vs this:
```php
<?php
$string = "text" . ('a' == 'a' ? '-append' : '');
echo $string;
```

Comment (Nicofuma) [inline phpBB/phpbb/search/fulltext_native.php]: I missed the concatenation actually. But please remove the extra parenthesis around the condition (not the whole ternary)

**Comment by marc1706** [comment]
This should probably still be replaced by sqlite3:
https://github.com/phpbb/phpbb/blob/3.2.x/phpBB/develop/create_schema_files.php#L28

**Pull request merged**